emloop/main_loop.py

```diff
@@ -299,9 +309,7 @@ def training():
                         hook.after_epoch(epoch_id=epoch_id, epoch_data=epoch_data)
 
                 for hook in self._hooks:
-                    hook.after_epoch_profile(epoch_id=epoch_id, profile=self._epoch_profile,
-                                             train_stream_name=self._train_stream_name,
-                                             extra_streams=self._extra_streams)
+                    hook.after_epoch_profile(epoch_id=epoch_id, profile=self._epoch_profile)
```

Comment:
I don't particularly like the stream extraction code in the `LogProfile` hook. To avoid that, I suggest to pass
```python
..., streams=[self._train_stream_name] + self._extra_streams)
```

Even better, I would probably like if the `LogProfile` hook did not need to parse the stream names whatsoever and instead, the profile is stored in a more reasonable form. E.g., instead of `batch_read_valid`, it would store a `dict` with keys such as `train`, `valid`, ... What does @blazekadam think?

Reply:
I do not like it either, please pass the stream names instead of parsing them as @FloopCZ suggests.

We can postpone restructuring the ``profile`` object for later.